go.mod: bump Pebble to 6f2788660198, rework shared storage wrapper

RaduBerinde opened the pull request:
6f278866 shared: improve interface for more efficient reading
9eb2c407 db: log events to testing.T in unit tests
f32e7dc6 db: add reserved Pebblev4 sstable format
5a6b91b8 objstorage: improve test and add read ahead test
2bc4319e objstorage: remove genericFileReadable
8143ffb9 objstorage: fix readaheadState initialization
06d08888 db: add Compact.Duration metric
e7213de0 db: add Uptime metric
e9005aed db: don't delete files during ingest application
222b43ec internal/arenaskl: fix Skiplist doc typo

Release note: None
Epic: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/103538)
<!-- Reviewable:end -->

RaduBerinde committed:
go.mod: bump Pebble to 6f2788660198, rework shared storage wrapper

6f278866 shared: improve interface for more efficient reading
9eb2c407 db: log events to testing.T in unit tests
f32e7dc6 db: add reserved Pebblev4 sstable format
5a6b91b8 objstorage: improve test and add read ahead test
2bc4319e objstorage: remove genericFileReadable
8143ffb9 objstorage: fix readaheadState initialization
06d08888 db: add Compact.Duration metric
e7213de0 db: add Uptime metric
e9005aed db: don't delete files during ingest application
222b43ec internal/arenaskl: fix Skiplist doc typo

Release note: None
Epic: none

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103538#-:-NVjtLYI5Z9hphSEr8ZK:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @itsbilal and @RaduBerinde)

___
*[`pkg/storage/shared_storage.go` line 51 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/103538#-NVjszZxCOlJIcajQcvW:-NVjszZxCOlJIcajQcvX:b3w1exo) ([raw file](https://github.com/cockroachdb/cockroach/blob/bc9288de280722619ad7427535c9c5b1e6e4d809/pkg/storage/shared_storage.go#L51)):*
> ```Go
> 		n += nn
> 	}
> 	atomic.AddInt64(&r.p.sharedBytesRead, int64(len(p)))
> ```

if we return up above, we may have read more than `0` bytes before returning. is it worth still incrementing `sharedBytesRead` in that case? 


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
*[`pkg/storage/shared_storage.go` line 51 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/103538#-NVjszZxCOlJIcajQcvW:-NVju7HJ7E-g2qADVebg:b-ixsxwl) ([raw file](https://github.com/cockroachdb/cockroach/blob/bc9288de280722619ad7427535c9c5b1e6e4d809/pkg/storage/shared_storage.go#L51)):*
<details><summary><i>Previously, jbowens (Jackson Owens) wrote…</i></summary><blockquote>

if we return up above, we may have read more than `0` bytes before returning. is it worth still incrementing `sharedBytesRead` in that case? 
</blockquote></details>

Could go either way here.. but any partial reads are not useful to the caller


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
